fastparquet/api.py

```diff
@@ -918,6 +918,20 @@ def __getstate__(self):
 
     def __setstate__(self, state):
         self.__dict__.update(state)
+        # Decode 'file_path'.
+        fmd = self.fmd
```

Comment:
in this case, you are not editing a list of objects, but attributes of an object deep in the list, so you can work on `self.fmd` directly without this line and `self.fmd = ` later on.